src/framework-ondata.js

```diff
@@ -63,7 +63,7 @@ module.exports = function(payload, socketIndex) {
 
              new Promise(function( resolve ) { 
                  // Execute fRunUpdater which will also execute the updater
-                 resolve( wrapper.fRunUpdater( vni ) );
+                 resolve( wrapper.fRunUpdater( vni, payload ) );
```

Comment:
It is being used in join. I developed the code at the same time, but split it up for review.   I'll remove it for now for the split review.